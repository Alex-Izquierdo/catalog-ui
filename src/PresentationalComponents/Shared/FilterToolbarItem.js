import React, { Component } from 'react';
import propTypes from 'prop-types';
import { Section } from '@red-hat-insights/insights-frontend-components';
import { ToolbarGroup, ToolbarItem, TolbarSection, Title, Button, TextInput } from '@patternfly/react-core';
import { css } from '@patternfly/react-styles';

class FilterToolbarItem extends Component {
    state = {
        searchValue: ''
    };

    render() {
        const { searchValue } = this.state;
        return (
            <ToolbarGroup className="searchToolbar">
                <ToolbarItem>
                    <div className="pf-c-input-group">
                        <TextInput placeholder={ this.props.placeholder } value={ searchValue } type="text" onChange={ this.props.onFilterChange }
                            aria-label="Find product button"></TextInput>
                        <Button variant="tertiary" id="searchProductButton" onClick={ this.onSearchSubmit }>
                            <i className="fas fa-search" aria-hidden="true"></i>
                        </Button>
                    </div>
                </ToolbarItem>
            </ToolbarGroup>);
    };
};

FilterToolbarItem.propTypes = {
    history: propTypes.object,
    onFilterChange: propTypes.func,
    placeholder: propTypes.string
};

export default FilterToolbarItem;

