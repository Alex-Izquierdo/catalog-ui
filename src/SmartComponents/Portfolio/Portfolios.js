import React, { Component, Fragment } from 'react';
import { connect } from 'react-redux';
import propTypes from 'prop-types';
import { Route, Link } from 'react-router-dom';
import { Toolbar, ToolbarGroup, ToolbarItem, Title, Button } from '@patternfly/react-core';
import { css } from '@patternfly/react-styles';
import spacingStyles from '@patternfly/patternfly-next/utilities/Spacing/spacing.css';
import flexStyles from '@patternfly/patternfly-next/utilities/Flex/flex.css';
import ContentGallery from '../../SmartComponents/ContentGallery/ContentGallery';
import PortfolioCard from '../../PresentationalComponents/Portfolio/PorfolioCard';
import PortfoliosFilterToolbar from '../../PresentationalComponents/Portfolio/PortfoliosFilterToolbar';
import { fetchPortfolios } from '../../redux/Actions/PortfolioActions';
import { hideModal, showModal } from '../../redux/Actions/MainModalActions';
import AddPortfolio from './add-portfolio-modal';
import './portfolio.scss';

class Portfolios extends Component {
    state = {
      filteredItems: [],
      isOpen: false
    };

    fetchData = () => {
      this.props.fetchPortfolios();
    };

    componentDidMount() {
      this.fetchData();
    }

    renderToolbar() {
      return (
        <Toolbar className={ css(flexStyles.justifyContentSpaceBetween, spacingStyles.mxXl, spacingStyles.myMd) }>
          <ToolbarGroup>
            <ToolbarItem className={ css(spacingStyles.mrXl) }>
              <Title size={ '2xl' }> All Portfolios</Title>
            </ToolbarItem>
          </ToolbarGroup>
          <ToolbarGroup  className={ 'pf-u-ml-auto-on-xl' }>
            <ToolbarItem>
              <Link to="/portfolios/add-portfolio">
                <Button
                  variant="primary"
                  aria-label="Create Portfolio"
                >
                Create Portfolio
                </Button>
              </Link>
            </ToolbarItem>
          </ToolbarGroup>
        </Toolbar>
      );
    }

    onClickCreatePortfolio = () => {
      this.props.showModal({
        open: true,
        closeModal: this.props.hideModal
      }, 'addportfolio');

      this.setState({
        ...this.state,
        isOpen: !this.state.isOpen
      });
    };

    render() {
      let filteredItems = {
        items: this.props.portfolios.map((item) => <PortfolioCard key={ item.id } { ...item } />),
        isLoading: this.props.isLoading
      };
      console.log('Portfolios foo');

      return (
        <Fragment>
          <PortfoliosFilterToolbar/>
          <Route exact path="/portfolios/add-portfolio" component={ AddPortfolio } />
          <Route exact path="/portfolios/edit/:id" component={ AddPortfolio } />
          { this.renderToolbar() }
          <ContentGallery { ...filteredItems } />
        </Fragment>
      );
    }
}

const mapStateToProps = ({ portfolioReducer: { portfolios, isLoading, filterValue }}) => ({
  portfolios,
  isLoading,
  searchFilter: filterValue
});

const mapDispatchToProps = dispatch => {
  return {
    fetchPortfolios: apiProps => dispatch(fetchPortfolios(apiProps)),
    hideModal: () => dispatch(hideModal()),
    showModal: (modalProps, modalType) => {
      dispatch(showModal({ modalProps, modalType }));
    }
  };
};

Portfolios.propTypes = {
  filteredItems: propTypes.array,
  portfolios: propTypes.array,
  platforms: propTypes.array,
  isLoading: propTypes.bool,
  searchFilter: propTypes.string,
  showModal: propTypes.func,
  hideModal: propTypes.func,
  fetchPortfolios: propTypes.func.isRequired
};

export default connect(mapStateToProps, mapDispatchToProps)(Portfolios);
