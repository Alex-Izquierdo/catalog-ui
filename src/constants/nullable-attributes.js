export const PORTFOLIO_ITEM_NULLABLE = [
  'display_name', 'description', 'long_description', 'distributor', 'documentation_url', 'support_url', 'workflow_ref'
];
