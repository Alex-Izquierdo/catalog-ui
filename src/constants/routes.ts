export const ADD_PORTFOLIO_ROUTE = '/portfolios/add-portfolio';
export const ADD_PRODUCTS_ROUTE = '/portfolios/add-products';
export const EDIT_ORDER_PROCESS_ROUTE = '/portfolios/edit-order-process';
export const EDIT_PORTFOLIO_ROUTE = '/portfolios/edit';
export const EDIT_PORTFOLIO_WORKFLOW_ROUTE = '/portfolios/edit-workflow';
export const EDIT_SURVEY_ROUTE = '/portfolios/portfolio-item/edit-survey';
export const NESTED_EDIT_ORDER_PROCESS_ROUTE =
  '/portfolios/portfolio/edit-order-process';
export const NESTED_EDIT_PORTFOLIO_ROUTE =
  '/portfolios/portfolio/edit-portfolio';
export const NESTED_REMOVE_PORTFOLIO_ROUTE =
  '/portfolios/portfolio/remove-portfolio';
export const NESTED_SHARE_PORTFOLIO_ROUTE =
  '/portfolios/portfolio/share-portfolio';
export const NESTED_WORKFLOW_PORTFOLIO_ROUTE =
  '/portfolios/portfolio/edit-workflow';
export const ORDER_ROUTE = '/orders/order';
export const ORDERS_ROUTE = '/orders';
export const PLATFORM_INVENTORIES_ROUTE = '/platforms/platform-inventories';
export const PLATFORM_ROUTE = '/platforms/platform';
export const PLATFORM_SERVICE_OFFERINGS_ROUTE = '/platforms/service-offerings';
export const PLATFORM_TEMPLATES_ROUTE = '/platforms/platform-templates';
export const PLATFORM_DETAILS_ROUTE = '/platforms/platform-details';
export const PLATFORMS_ROUTE = '/platforms';
export const PORTFOLIO_ROUTE = '/portfolios/portfolio';
export const PORTFOLIO_ITEM_ROUTE = '/portfolios/portfolio-item';
export const PORTFOLIO_ITEM_ORDER_ROUTE = '/portfolios/portfolio-item/order';
export const PORTFOLIO_ITEM_ROUTE_EDIT = '/portfolios/portfolio-item/edit';
export const PORTFOLIO_ITEM_EDIT_ORDER_PROCESS_ROUTE =
  '/portfolios/portfolio-item/edit-order-process';
export const PORTFOLIOS_ROUTE = '/portfolios';
export const REMOVE_PORTFOLIO_ROUTE = '/portfolios/remove';
export const SHARE_PORTFOLIO_ROUTE = '/portfolios/share';
export const WORKFLOW_PORTFOLIO_ROUTE = '/portfolios/edit-workflow';
export const ORDER_PROCESSES_ROUTE = '/order-processes';
export const ADD_ORDER_PROCESS_ROUTE = '/order-processes/add-order-process';
export const REMOVE_ORDER_PROCESS_ROUTE = '/order-processes/remove';
export const UPDATE_ORDER_PROCESS_ROUTE = '/order-processes/update';
