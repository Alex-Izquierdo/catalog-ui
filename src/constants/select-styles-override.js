const style = {
  container: provided => ({
    ...provided,
    '&:focus': {
      outline: 'none'
    }
  }),
  control: (provided, { isFocused }) => ({
    ...provided,
    borderRadius: 0,
    borderColor: '#EDEDED',
    backgroundColor: '#FFFFFF',
    borderBottomColor: isFocused ? '#007bba' : '#282d33',
    cursor: 'pointer',
    boxShadow: 'none',
    minHeight: 36,
    width: 256,
    '&:hover': {
      borderBottomColor: '#007bba'
    }
  }),
  indicatorSeparator: () => ({}),
  dropdownIndicator: provided => ({
    ...provided,
    color: '#282d33',
    paddingBottom: 6,
    paddingTop: 6
  }),
  menu: provided => ({
    ...provided,
    borderRadius: 'none',
    boxShadow:
      '0 0.0625rem 0.0625rem 0rem rgba(3,3,3,.05),0 0.25rem 0.5rem 0.25rem rgba(3,3,3,.06)',
    paddingBottom: '0.5rem',
    paddingTop: '0.5rem',
    marginTop: 4,
    width: 'calc(100% - 4px)',
    marginRight: 2,
    marginLeft: 2
  }),
  option: (provided, { isFocused }) => ({
    ...provided,
    cursor: 'pointer',
    textAlign: 'left',
    color: '#282d33',
    fontSize: '1rem',
    fontWeight: '400',
    lineHeight: '1.5',
    paddingBottom: '0.5rem',
    paddingTop: '0.5rem',
    paddingLeft: '1rem',
    paddingRight: '1rem',
    backgroundColor: isFocused ? '#ededed' : 'transparent',
    '&:hover': {
      backgroundColor: '#ededed'
    }
  }),
  placeholder: provided => ({
    ...provided,
    left: 20
  }),
  singleValue: provided => ({
    ...provided,
    left: 20,
    maxWidth: 'calc(100% - 28px)'
  })
};

export default style;
