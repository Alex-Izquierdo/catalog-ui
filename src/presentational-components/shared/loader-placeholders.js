import React, { Fragment } from 'react';
import ContentLoader, { List } from 'react-content-loader';
import PropTypes from 'prop-types';
import { Main } from '@redhat-cloud-services/frontend-components/components/Main';
import { Spinner } from '@redhat-cloud-services/frontend-components/components/Spinner';
import {
  Bullseye,
  Card,
  CardBody,
  DataList,
  DataListCell,
  DataListItem,
  DataListItemRow,
  DataListItemCells,
  Grid,
  GridItem,
  Gallery,
  GalleryItem,
  Form,
  FormGroup,
  TextContent,
  Text,
  TextVariants
} from '@patternfly/react-core';

export const CardLoader = ({ items, ...props }) => (
  <Grid gutter="md">
    <GridItem sm={12} className="pf-u-p-md">
      <Gallery gutter="md">
        {[...Array(items)].map((_item, index) => (
          <GalleryItem key={index}>
            <Card style={{ height: 350 }}>
              <CardBody>
                <ContentLoader
                  height={160}
                  width={300}
                  speed={2}
                  primaryColor="#f3f3f3"
                  secondaryColor="#ecebeb"
                  {...props}
                >
                  <rect x="2" y="99" rx="3" ry="3" width="300" height="6.4" />
                  <rect
                    x="2"
                    y="119.72"
                    rx="3"
                    ry="3"
                    width="290"
                    height="5.76"
                  />
                  <rect x="2" y="139" rx="3" ry="3" width="201" height="6.4" />
                  <rect
                    x="-2.16"
                    y="0.67"
                    rx="0"
                    ry="0"
                    width="271.6"
                    height="82.74"
                  />
                  <rect
                    x="136.84"
                    y="37.67"
                    rx="0"
                    ry="0"
                    width="6"
                    height="3"
                  />
                </ContentLoader>
              </CardBody>
            </Card>
          </GalleryItem>
        ))}
      </Gallery>
    </GridItem>
  </Grid>
);

CardLoader.propTypes = {
  items: PropTypes.number
};

CardLoader.defaultProps = {
  items: 13
};

export const PortfolioLoader = ({ items, ...props }) => (
  <Grid gutter="md">
    <GridItem sm={12}>
      <ContentLoader
        height={16}
        width={300}
        speed={2}
        primaryColor="#FFFFFF"
        secondaryColor="#FFFFFF"
        {...props}
      >
        <rect x="0" y="0" rx="0" ry="0" width="420" height="16" />
      </ContentLoader>
    </GridItem>
    <GridItem sm={12} style={{ paddingLeft: 16, paddingRight: 16 }}>
      <CardLoader items={items} />
    </GridItem>
  </Grid>
);

PortfolioLoader.propTypes = {
  items: PropTypes.number
};

PortfolioLoader.defaultProps = {
  items: 5
};

export const AppPlaceholder = (props) => (
  <Main className="pf-u-m-0 pf-u-p-0">
    <ContentLoader
      height={16}
      width={300}
      speed={2}
      primaryColor="#FFFFFF"
      secondaryColor="#FFFFFF"
      {...props}
    >
      <rect x="0" y="0" rx="0" ry="0" width="420" height="10" />
    </ContentLoader>
    <div>
      <Bullseye>
        <Spinner />
      </Bullseye>
    </div>
  </Main>
);

export const ToolbarTitlePlaceholder = (props) => (
  <ContentLoader
    height={21}
    width={200}
    speed={2}
    primaryColor="#f3f3f3"
    secondaryColor="#ecebeb"
    {...props}
  >
    <rect x="0" y="0" rx="0" ry="0" width="200" height="21" />
  </ContentLoader>
);

export const ProductLoaderPlaceholder = (props) => (
  <Fragment>
    <ContentLoader
      height={15}
      width={200}
      speed={2}
      primaryColor="#f3f3f3"
      secondaryColor="#ecebeb"
      {...props}
    >
      <rect x="0" y="0" rx="0" ry="0" width="200" height="10" />
    </ContentLoader>
    <div style={{ width: 300 }}>
      <List />
      <List speed={3} />
      <List />
    </div>
  </Fragment>
);

export const IconPlaceholder = ({ height, ...props }) => (
  <svg height={height} width={height}>
    <circle cx={height / 2} cy={height / 2} r={height / 2} fill="#ecebeb" />
  </svg>
);

IconPlaceholder.propTypes = {
  height: PropTypes.oneOfType([PropTypes.string, PropTypes.number])
};

IconPlaceholder.defaultProps = {
  height: '40'
};

const FormItemLoader = () => (
  <ContentLoader
    height={36}
    width={400}
    speed={2}
    primaryColor="#ffffff"
    secondaryColor="#ecebeb"
  >
    <rect x="0" y="0" rx="0" ry="0" width="400" height="36" />
  </ContentLoader>
);

export const ShareLoader = () => (
  <Form>
    <TextContent>
      <Text component={TextVariants.small}>Invite group</Text>
    </TextContent>
    <FormGroup fieldId="1">
      <FormItemLoader />
    </FormGroup>
    <TextContent>
      <Text component={TextVariants.small}>Groups with access</Text>
    </TextContent>
    <FormGroup fieldId="3">
      <FormItemLoader />
    </FormGroup>
    <FormGroup fieldId="4">
      <FormItemLoader />
    </FormGroup>
  </Form>
);

export const WorkflowLoader = () => (
  <Form>
    <FormGroup fieldId="1">
      <TextContent>
        <Text component={TextVariants.medium}>Approval workflow</Text>
      </TextContent>
    </FormGroup>
    <FormGroup fieldId="2">
      <FormItemLoader />
    </FormGroup>
  </Form>
);

export const ListLoader = ({ items, ...props }) => (
  <DataList aria-label="list-loader" aria-labelledby="datalist-placeholder">
    {[...Array(items)].map((_item, index) => (
      <DataListItem key={index} aria-labelledby="datalist-item-placeholder">
        <DataListItemRow aria-label="datalist-item-placeholder-row">
          <DataListItemCells
            dataListCells={[
              <DataListCell key="1">
                <ContentLoader
                  height={12}
                  width={300}
                  speed={2}
                  primaryColor="#FFFFFF"
                  secondaryColor="#ecebeb"
                  {...props}
                >
                  <rect x="0" y="0" rx="0" ry="0" width="300" height="12" />
                </ContentLoader>
              </DataListCell>
            ]}
          />
        </DataListItemRow>
      </DataListItem>
    ))}
  </DataList>
);

ListLoader.propTypes = {
  items: PropTypes.number
};

ListLoader.defaultProps = {
  items: 5
};

export const OrderDetailToolbarPlaceholder = () => (
  <div>
    <ContentLoader height={20} width={300}>
      <rect x="0" y="0" rx="0" ry="0" width="300" height="12" />
    </ContentLoader>
  </div>
);
