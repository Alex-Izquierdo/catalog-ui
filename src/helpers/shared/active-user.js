/* eslint camelcase: 0 */
import { getAxiosInstance } from '../shared/user-login';
import { CATALOG_API_BASE } from '../../utilities/constants';
const axiosInstance = getAxiosInstance();

export const getUser = () => axiosInstance.get(`${CATALOG_API_BASE}/me/`);
