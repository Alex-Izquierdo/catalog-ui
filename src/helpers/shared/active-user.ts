/* eslint camelcase: 0 */
import { getAxiosInstance } from '../shared/user-login';
import {
  CATALOG_API_BASE,
  EXTERNAL_LOGIN_URI
} from '../../utilities/constants';
const axiosInstance = getAxiosInstance();

export const getUser = () => axiosInstance.get(`${CATALOG_API_BASE}/me/`);

export const logoutUser = () => {
  localStorage.removeItem('catalog_user');
  return axiosInstance.post(`${CATALOG_API_BASE}/logout/`);
};

export const loginUser = () => {
  window.location.replace(EXTERNAL_LOGIN_URI);
  return null;
};
