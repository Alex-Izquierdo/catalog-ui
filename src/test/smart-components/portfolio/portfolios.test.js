import React from 'react';
import { act } from 'react-dom/test-utils';
import thunk from 'redux-thunk';
import { shallow, mount } from 'enzyme';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store' ;
import { shallowToJson } from 'enzyme-to-json';
import { MemoryRouter, Route } from 'react-router-dom';
import promiseMiddleware from 'redux-promise-middleware';
import { notificationsMiddleware } from '@redhat-cloud-services/frontend-components-notifications/';

import { CATALOG_API_BASE } from '../../../utilities/constants';
import { FETCH_PORTFOLIOS } from '../../../redux/action-types';
import Portfolios from '../../../smart-components/portfolio/portfolios';
import PortfolioCard from '../../../presentational-components/portfolio/porfolio-card';
import { CardLoader } from '../../../presentational-components/shared/loader-placeholders';
import FilterToolbarItem from '../../../presentational-components/shared/filter-toolbar-item';

describe('<Portfolios />', () => {
  let initialProps;
  let initialState;
  const middlewares = [ thunk, promiseMiddleware(), notificationsMiddleware() ];
  let mockStore;

  const ComponentWrapper = ({ store, initialEntries = [ '/foo' ], children }) => (
    <Provider store={ store }>
      <MemoryRouter initialEntries={ initialEntries }>
        { children }
      </MemoryRouter>
    </Provider>
  );

  beforeEach(() => {
    initialProps = {
      id: '123'
    };
    initialState = {
      portfolioReducer: {
        portfolios: { data: [{
          id: '123',
          name: 'bar',
          description: 'description',
          modified: 'sometimes',
          created_at: 'foo',
          owner: 'Owner'
        }],
        meta: {
          limit: 50,
          offset: 0
        }}
      }
    };
    mockStore = configureStore(middlewares);
  });

  it('should render correctly', () => {
    const store = mockStore(initialState);
    const wrapper = shallow(
      <ComponentWrapper store={ store } initialEntries={ [ '/portfolios' ] }>
        <Portfolios { ...initialProps } store={ store }/>
      </ComponentWrapper>).find(Portfolios);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  it('should mount and fetch data', (done) => {
    const store = mockStore(initialState);

    apiClientMock.get(`${CATALOG_API_BASE}/portfolios?limit=50&offset=0`, mockOnce({ body: { data: [{ name: 'Foo', id: '11' }]}}));
    const expectedActions = [{
      type: `${FETCH_PORTFOLIOS}_PENDING`
    }, expect.objectContaining({
      type: `${FETCH_PORTFOLIOS}_FULFILLED`
    }) ];
    mount(
      <ComponentWrapper store={ store } initialEntries={ [ '/portfolios' ] }>
        <Route path="/portfolios" render={ args => <Portfolios { ...initialProps } { ...args } /> } />
      </ComponentWrapper>
    );

    setImmediate(() => {
      expect(store.getActions()).toEqual(expectedActions);
      done();
    });
  });

  it('should mount filter portfolios', (done) => {
    const store = mockStore(initialState);

    apiClientMock.get(`${CATALOG_API_BASE}/portfolios?limit=50&offset=0`, mockOnce({ body: { data: [{ name: 'Foo', id: '11' }]}}));

    const wrapper = mount(
      <ComponentWrapper store={ store } initialEntries={ [ '/portfolios' ] }>
        <Route path="/portfolios" render={ args => <Portfolios { ...initialProps } { ...args } /> } />
      </ComponentWrapper>
    );

    setImmediate(() => {
      wrapper.update();
      expect(wrapper.find(PortfolioCard)).toHaveLength(1);
      const filterInput = wrapper.find(FilterToolbarItem).first();
      filterInput.props().onFilterChange('nothing');
      wrapper.update();
      expect(wrapper.find(PortfolioCard)).toHaveLength(0);
      done();
    });
  });

  it('should render in loading state', async (done) => {
    const store = mockStore({
      ...initialState,
      portfolioReducer: {
        ...initialState.portfolioReducer,
        isLoading: true,
        portfolios: { data: [], meta: { limit: 50, offset: 0 }}
      }
    });

    apiClientMock.get(`${CATALOG_API_BASE}/portfolios?limit=50&offset=0`, mockOnce({ body: { data: [{ name: 'Foo', id: '11' }]}}));
    let wrapper;
    await act(async () => {
      wrapper = mount(
        <ComponentWrapper store={ store } initialEntries={ [ '/portfolios' ] }>
          <Route exact path="/portfolios" render={ props => <Portfolios { ...initialProps } { ...props } /> } />
        </ComponentWrapper>
      );
    });

    setImmediate(() => {
      expect(wrapper.find(CardLoader)).toHaveLength(1);
      done();
    });
  });
});
