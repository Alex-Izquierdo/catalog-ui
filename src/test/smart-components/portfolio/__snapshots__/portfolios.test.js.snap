// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Portfolios /> should render correctly 1`] = `
<ConnectFunction
  id="123"
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
/>
`;
